### PR TITLE
Add support for building on Little Endian PPC64

### PR DESCRIPTION
**Pull request by amock**
This should fix https://github.com/briansmith/ring/issues/1567

### CODE REVIEW

**Comment by ecnelises** [comment]
There is https://github.com/briansmith/ring/pull/1057 (and maybe other) for enabling ppc64le, and more work at https://github.com/IBM/ring/commits/ppc-0.16.20. These (including patches for other archs) are pending on review.

**Review by runlevel5 — APPROVED** [review]
LGTM 👍

File: build.rs [hunk 1]
```diff
@@ -45,7 +46,7 @@ const RING_SRCS: &[(&[&str], &str)] = &[
     (&[], "crypto/mem.c"),
     (&[], "crypto/poly1305/poly1305.c"),
 
-    (&[AARCH64, ARM, X86_64, X86], "crypto/crypto.c"),
```
Comment (briansmith) [inline build.rs:48]: This isn't needed.

File: include/ring-core/base.h [hunk 2]
```diff
@@ -89,6 +89,9 @@
 #elif defined(__MIPSEL__) && defined(__LP64__)
 #define OPENSSL_64_BIT
 #define OPENSSL_MIPS64
+#elif defined(__PPC64__) && defined(__LITTLE_ENDIAN__)
+#define OPENSSL_64_BIT
+#define OPENSSL_NO_ASM
```
Comment (briansmith) [inline include/ring-core/base.h:94]: I hope that `OPENSSL_NO_ASM` isn't needed because we don't define it for any other targets.

File: build.rs [hunk 3]
```diff
@@ -30,6 +30,7 @@ const X86: &str = "x86";
 const X86_64: &str = "x86_64";
 const AARCH64: &str = "aarch64";
 const ARM: &str = "arm";
+const PPC64LE: &str = "ppc64le";
```
Comment (briansmith) [inline build.rs:33]: This is also not needed.

**Pull request closed without merging**